src/main/allowProtocolDialog.js

```diff
@@ -37,51 +37,49 @@ function init(mainWindow) {
         protocol.schemes.forEach(addScheme);
       }
     });
-    initDialogEvent(mainWindow);
   });
 }
 
-function initDialogEvent() {
-  ipcMain.on('confirm-protocol', (event, protocol, URL) => {
-    if (allowedProtocols.indexOf(protocol) !== -1) {
-      shell.openExternal(URL);
-      return;
-    }
-    dialog.showMessageBox(getMainWindow(), {
-      title: 'Non http(s) protocol',
-      message: `${protocol} link requires an external application.`,
-      detail: `The requested link is ${URL} . Do you want to continue?`,
-      type: 'warning',
-      buttons: [
-        'Yes',
-        `Yes (Save ${protocol} as allowed)`,
-        'No',
-      ],
-      cancelId: 2,
-      noLink: true,
-    }).then(({response}) => {
-      switch (response) {
-      case 1: {
-        allowedProtocols.push(protocol);
-        function handleError(err) {
-          if (err) {
-            console.error(err);
-          }
+function handleDialogEvent(protocol, URL) {
+  if (allowedProtocols.indexOf(protocol) !== -1) {
```

Comment:
is this a save enough check? What if another protocol is removed or this ends up being at this position?

Reply:
`indexOf` only returns `-1` if the requested item isn't in the array, so it will only pass this check if the protocol is in `allowedProtocols`.

`allowedProtocols` is a singleton in this context, meaning that there will only ever be one instance of it in the app, and it's only modified by this function or upon loading the initial list of allowed protocols.